app/controllers/volunteers_controller.rb

```diff
@@ -50,9 +50,8 @@ def inactive
   end
 
   def need_training
-    @volunteers = Volunteer.all.keep_if do |volunteer|
-      !(volunteer.region_ids & current_volunteer.region_ids).empty? && volunteer.needs_training?
-    end
+    @volunteers = Volunteer.not_super_admin.assigned_to_regions(current_volunteer.region_ids).select { |v| v.needs_training? }
```

Comment:
```
@volunteers = Volunteer.not_super_admin
                       .assigned_to_regions(current_volunteer.region_ids)
                       .select { |v| v.needs_training? }
```
?

I think there is also a scope now called `in_regions` that you'll see when you merge in master.